drivers/timer/counter_timer.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <counter.h>
+#include <drivers/system_timer.h>
+
+extern int z_clock_hw_cycles_per_sec;
+
+static struct device *counter;	/* Counter device used as clock source. */
+static u32_t curr_cycle;	/* Current (last announced) clock cycle. */
+static u32_t alarm_cycle;	/* Clock cycle on which alarm is scheduled. */
+
+static void alarm(struct device *dev, u8_t chan_id,
+		  u32_t cycle, void *user_data)
+{
+	s32_t ticks;
+
+	ticks = (cycle - curr_cycle) / sys_clock_hw_cycles_per_tick();
+	curr_cycle = cycle;
```

Comment:
if ticks is zero we should do nothing and leave.
or is it guaranteed not to happen?